game-core/src/main/java/games/strategy/engine/config/PropertyReader.java

```diff
@@ -15,4 +20,68 @@
    * @throws IllegalStateException If an error occurs reading the property source.
    */
   String readProperty(String key);
+
+  /**
+   * Reads the specified property or uses the specified default value if the property does not exist.
+   *
+   * @param key The property key.
+   * @param defaultValue The default property value.
+   *
+   * @return The property value or {@code defaultValue} if the key is not present.
+   *
+   * @throws IllegalArgumentException If {@code key} is empty or contains only whitespace.
+   * @throws IllegalStateException If an error occurs reading the property source.
+   */
+  default String readPropertyOrDefault(final String key, final String defaultValue) {
+    checkNotNull(defaultValue);
+
+    final String value = readProperty(key);
+    return !value.isEmpty() ? value : defaultValue;
+  }
+
+  /**
+   * Reads the specified boolean property or uses the specified default value if the property does not exist.
+   *
+   * @param key The property key.
+   * @param defaultValue The default property value.
+   *
+   * @return The boolean property value or {@code defaultValue} if the key is not present.
+   *
+   * @throws IllegalArgumentException If {@code key} is empty or contains only whitespace.
+   * @throws IllegalStateException If an error occurs reading the property source.
+   */
+  default boolean readBooleanPropertyOrDefault(final String key, final boolean defaultValue) {
+    final String value = readProperty(key);
+    return !value.isEmpty() ? Boolean.parseBoolean(value) : defaultValue;
+  }
+
+  /**
+   * Reads the specified integer property or uses the specified default value if the property does not exist.
+   *
+   * @param key The property key.
+   * @param defaultValue The default property value.
+   *
+   * @return The integer property value or {@code defaultValue} if the key is not present or if the property value
+   *         cannot be parsed as an integer.
+   *
+   * @throws IllegalArgumentException If {@code key} is empty or contains only whitespace.
+   * @throws IllegalStateException If an error occurs reading the property source.
+   */
+  default int readIntegerPropertyOrDefault(final String key, final int defaultValue) {
+    final String value = readProperty(key);
+    if (!value.isEmpty()) {
+      try {
+        return Integer.parseInt(value);
+      } catch (final NumberFormatException e) {
+        Logger.getLogger(PropertyReader.class.getName()).log(
```

Comment:
I'm using a `Logger` here because this code can be called from both the client and the lobby, so `ClientLogger` is not appropriate.